internal/handler/msg_getmore.go

```diff
@@ -266,3 +273,68 @@ func (h *Handler) makeNextBatch(c *cursor.Cursor, batchSize int64) (*types.Array
 
 	return nextBatch, nil
 }
+
+// awaitData stops the goroutine, and waits for a new data for the cursor.
+// If there's a new document, or the maxTimeMS have passed it returns the nextBatch.
+func (h *Handler) awaitData(
+	ctx context.Context,
+	c *cursor.Cursor,
+	maxTimeMS, batchSize int64,
+) (resBatch *types.Array, err error) {
```

Comment:
```suggestion
//
//nolint:lll // for readability
func (h *Handler) awaitData(ctx context.Context, c *cursor.Cursor, maxTimeMS, batchSize int64) (resBatch *types.Array, err error) {
```

or use a `struct` for parameters.

Please ack, resolve, and tweak in the next PR